packages/go_router/lib/src/match.dart

```diff
@@ -218,7 +218,17 @@ abstract class RouteMatchBase with Diagnosticable {
     final String newMatchedLocation =
         concatenatePaths(matchedLocation, pathLoc);
     final String newMatchedPath = concatenatePaths(matchedPath, route.path);
-    if (newMatchedLocation.toLowerCase() == uri.path.toLowerCase()) {
+
+    final String caseSensitiveNewMatchedLocation;
```

Comment:
I think the new name is more confusing than the old name, because it can contains case insensitive string in the else case

Reply:
Fair enough, What do you think of `newMatchedLocationToCompare` and `uriPathToCompare` ? I added them in [Add better name and documentation](https://github.com/flutter/packages/pull/8992/commits/48ba2860c7edc3143ee53d5590aa35f04b1982d4)